src/server/auth.ts

```diff
@@ -66,31 +108,7 @@ export const createStrategyHandler = (authType: AuthType) => async (
 			}
 		} else {
 			// no user found, create new user
-			logger.debug('> Creating user.....');
-			const newUser = {
-				authLevel: AuthLevel.HACKER,
-				authType,
-				email: profile.emails[0].value,
-				googleId: profile.id,
-				password: `${authType}!123`,
-			};
-			const createdUser = await UserModel.create(newUser);
-			if (createdUser) {
-				// create hacker
-				const createdHacker = await HackerModel.create({
-					email: createdUser.email,
-					status: Status.Created,
-					user: createdUser._id,
-				});
-				if (createdHacker) {
-					logger.debug(createdUser);
-					done(null, createdUser);
-				} else {
-					done(null, false);
-				}
-			} else {
-				done(null, false);
-			}
+			createAuthenticatedUser(authType, profileEmail, (user as any).id, done);
```

Comment:
why as any?

Reply:
typescript says it can be null even though the else statement makes that impossible

Reply:
Couldn't you make it `User | null`

Reply:
`User | null` doesn't have the id field

Reply:
it doesn't matter in this case because we only care about the id, the user type is enforced when it's originally passed in as a param of the func

Reply:
You can return after the done to fake out typescript but also make it clearer to the reader of the code